### PR TITLE
[CS] refactor: update github actions

### PR DESCRIPTION
**Pull request by Anakael**
As discussed in: https://github.com/OpenAPITools/openapi-generator/pull/17471#discussion_r1446844543

Here what I come up with: 
samples-dotnet-standard.yaml -- for .netstandard clients
samples-dotnet.yaml -- for .net7+
samples-dotnet6-client-echo-api.yaml -- for client echo api
samples-dotnet6-client.yaml -- for .net6 clients and older (but not .netstandard)
samples-dotnet6-server.yaml -- for server tests (.net6+ only. Older versions can also be added, but its out of support now)

I've also moved one restsharp net7 client to fit general structure (client/petstore/csharp/**.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by Anakael** [comment]
@mandrean @shibayan @Blackclaws @lucamazzanti @iBicha @devhl-labs

File: .github/workflows/samples-dotnet-standard.yaml [hunk 1]
```diff
@@ -15,8 +27,13 @@ jobs:
...
         sample:
-          # clients
-          - samples/client/petstore/csharp/OpenAPIClient-generichost-netstandard2.0
+          # build C# API client (netstandard)
+          - samples/client/petstore/csharp/OpenAPIClient/**
+          - samples/client/petstore/csharp/*netstandard*/**
+          # build C# API client (netstandard with ConditionalSerialization)
+          - samples/client/petstore/csharp/OpenAPIClient-ConditionalSerialization/**
+          # build C# API client (netstandard httpclient)
+          - samples/client/petstore/csharp/OpenAPIClient-httpclient/**
```
Comment (wing328) [inline .github/workflows/samples-dotnet-standard.yaml:36]: for `sample` under matrix, I thought we must use the full path without using **. Am I right?

Comment (Anakael) [reply .github/workflows/samples-dotnet-standard.yaml]: Oh, I've just copied from top.
Done.

**Comment by Anakael** [comment]
@wing328 
Can we merge it? :)

**Comment by wing328** [comment]
thanks for the PR. let's give it a try

**Pull request merged**